src/main/java/Duke.java

```diff
@@ -1,10 +1,53 @@
+import java.io.FileNotFoundException;
+
 public class Duke {
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Attempts to create a TaskList object which contains tasks that are previously
+     * saved in the tasks.txt file. If the tasks.txt file cannot be read, create a
+     * new TaskList object.
+     *
+     * @param filePath File path to tasks.txt which contains saved tasks
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.getSavedTasks(ui));
+        } catch (FileNotFoundException e) {
+            ui.showFileLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Starts the Duke ChatBot and continuously prompts the user for
+     * commands until the user gives a "bye" command.
+     */
+    public void run() {
+        ui.showGreetings();
+        boolean shouldExitChat = false;
+        while (!shouldExitChat) {
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine();
+                Parser p = new Parser(fullCommand);
+                Command c = p.parse();
+                c.execute(tasks, ui, storage);
+                shouldExitChat = c.shouldExit();
+            } catch (DukeException e) {
+                ui.showCommandError();
+            } finally {
+                ui.showLine();
+            }
+        }
+
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        new Duke("data/tasks.txt").run();
```

Comment:
Avoid the usage of magic literals. Maybe declare a constant (static final var) and then utilise it. [See here](https://nus-cs2113-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers)